### ISSUE TITLE
NodeSelector value should be always string (Failed to deploy ingress-nginx, kube-dns, coredns when specify "XXX.com: true" in nodeSelector

### ISSUE THREAD

**Issue by ukinau** [rancher/rke#1437]
**RKE version:**
master

**cluster.yml file:**

```
ingress:
    provider: nginx
    node_selector:
      ingress-node.test.com: "true"
    options:
      proxy-read-timeout: "3600"
    extra_args:
      v: 2

addon_job_timeout: 30
authentication: 
  strategy: "x509"
ignore_docker_version: true

ingress: 
  provider: none
kubernetes_version: "v1.13.5-rancher1-2"
monitoring: 
  provider: "metrics-server"

network: 
  options: 
    flannel_backend_type: "vxlan"
  plugin: "flannel"

services: 
  etcd: 
    backup_config: 
      enabled: true
      interval_hours: 12
      retention: 6
    creation: "12h"
    extra_args: 
      heartbeat-interval: 500
      election-timeout: 5000
    retention: "72h"
    snapshot: false
  kube-api: 
    always_pull_images: false
    pod_security_policy: false
    service_node_port_range: "30000-32767"
    service_cluster_ip_range: 172.30.0.0/15
  kube-controller:
    cluster_cidr: 172.16.0.0/14
    service_cluster_ip_range: 172.30.0.0/15
  kubelet:
    cluster_domain: cluster.local
    cluster_dns_server: 172.30.0.10

ssh_agent_auth: false
docker_root_dir: "/var/lib/docker"
enable_cluster_alerting: false
enable_cluster_monitoring: false
enable_network_policy: false
local_cluster_auth_endpoint: 
  enabled: false
```

**Steps to Reproduce:**
 - prepare node which have `ingress-node.test.com: "true"` label
 - prepare cluster.yml which configure ingress.nodeSelector to use `ingress-node.test.com: "true"`
 - Run `rke up`

**Results:**
Continuously failed to complete `rke-ingress-controller-deploy-job`. 

**Investigation:**
Kubernetes always expect nodeSelector to have `string` as a value, but if we specified `true` without quote as a value,  `true` will be treated as a boolean and then kube-apiserver reject that request because of type mis-match, that's why we need to quote value in nodeSelector in template. Related issue https://github.com/kubernetes/kubernetes/issues/23251

**Comment by jiaqiluo** [comment]
The bug is reproduced in rke version v0.2.6

Steps:
- run rke up to start a single-node cluster
- add the label `ingress-node.test.com: "true"` to the node
- edit the `cluster.yml` file to append the following part
```
services:
  etcd:
    snapshot: true
    creation: 6h
    retention: 24h
 
ingress:
  provider: nginx
  node_selector:
    ingress-node.test.com: "true"
  options:
    proxy-read-timeout: "3600"
  extra_args:
    v: 2
```
- run `rke up`

result:
- the cluster is up, bit the output shows this error
```
WARN[0119] Failed to deploy addon execute job [rke-ingress-controller]: Failed to get job complete status for job rke-ingress-controller-deploy-job in namespace kube-system
```

**Comment by jiaqiluo** [comment]
The bug fix is validated in rke version v0.3.0-rc5

Following the same steps as the above comment, this time there is no warning message showing and the cluster is up.

Also, the deploy job is completed successfully.
```
kube-system     rke-ingress-controller-deploy-job-r8xzp   0/1     Completed     0          12s
```
